### PR TITLE
Add CUDA PPISP Regularization Loss 

### PR DESCRIPTION
**Pull request by shiaoszuc**
- Introduced a new custom autograd function for PPISP regularization loss in `__init__.py`.
- Updated the `PPISP` class to utilize the new regularization function for loss computation.
- Added CUDA kernel implementations for forward and backward passes of the regularization loss in `ppisp_impl.cu`.
- Defined necessary bindings in `ext.cpp` and updated header files to include new functions.
- Created a new test file `test_regularization_loss.py` to validate the correctness of the regularization loss against a PyTorch reference implementation.
- Updated README to include information about the new regularization loss tests.

Experiments on both **H100** and **RTX 6000 Pro** of our internal 3dgut-based training, which has 40000 training steps with 1 epoch

# H100
## Wall Time
Improve from **6385 seconds** to **5958 seconds**
## Loss Time Improvements

- 0 - 999 steps: 283.00us vs 6528.84us
- 10000 - 10999 steps: 255.34us vs 5596.33us
- 20000 - 20999 steps: 261.52us vs 6143.08us
- 30000 - 30999 steps: 290.52us vs 7410.24us

## Quality Comparison
- PSNR: 27.841677 (cuda) vs 27.799282 (torch)
- SSIM: 0.881647 (cuda) vs 0.881447 (torch)
- LPIPS: 0.191591 (cuda) vs 0.192679 (torch)

# RTX 6000 Pro
## Wall Time
Improve from **4158 seconds** to **3808 seconds**
## Loss Time Improvements

- 0 - 999 steps: 154.13us vs 4365.15us
- 10000 - 10999 steps: 139.25us vs 4672.14us
- 20000 - 20999 steps: 130.87us vs 5479.88 us
- 30000 - 30999 steps: 145.82 us vs 7298.04 us

## Quality Comparison
- PSNR: 27.804377 (cuda) vs 27.810232 (torch)
- SSIM: 0.881480 (cuda) vs 0.880953 (torch)
- LPIPS: 0.191799 (cuda) vs 0.192077 (torch)

### CODE REVIEW

**Comment by moennen** [comment]
@shiaoszuc : main and only comment : we should get rid of the torch path, let's avoid all the clutter with the optionnal argument that nobody is going to set anyway. :pray:

**Comment by moennen** [comment]
@shiaoszuc : could you also put in the MR description the results (PSNR, SSIM and training time) of running an optimization before and after the switch to cuda ? :pray:

**Comment by nvibd** [comment]
Thank you for this PR! I will run a benchmark on our datasets to ensure the metrics hold.

**Comment by nvibd** [comment]
Please bump the library version in `_version.py`.

**Comment by shiaoszuc** [comment]
> @shiaoszuc : main and only comment : we should get rid of the torch path, let's avoid all the clutter with the optionnal argument that nobody is going to set anyway. 🙏

done

> Please bump the library version in `_version.py`.

should I bump to 1.1.3 or 1.2.0 ?

**Comment by shiaoszuc** [comment]
> @shiaoszuc : could you also put in the MR description the results (PSNR, SSIM and training time) of running an optimization before and after the switch to cuda ? 🙏

done
